services/py-api/tests/unit_tests/conftest.py

```diff
@@ -54,6 +56,9 @@ def close_side_effect() -> Optional[Err[str]]:
 
 @pytest.fixture
 def participant_repo_mock() -> Mock:
+    """This is a mock obj of ParticipantsRepository. To change the return values of its methods use:
+    `participant_repo_mock.method_name.return_value=some_return_value`"""
+
```

Comment:
Nice clarification! Question for you: Although we are assigning these methods Mock objects, do these object contain the implementation of the original methods, or you can only bind them with return values?

Reply:
Mock objects are mock objects, meaning fake. The idea is that you don't want to call the original method. So, no, they do not contain the implementation of the original methods. they are just fake properties/methods called during the execution of the test. And you can make them return whatever you want or add side effects, for example, throwing exceptions.